crates/ruff/src/rules/flake8_comprehensions/fixes.rs

```diff
@@ -552,7 +573,51 @@ pub(crate) fn fix_unnecessary_collection_call(
     };
     tree.codegen(&mut state);
 
-    Ok(Edit::range_replacement(state.to_string(), expr.range()))
+    let code = if wrap_in_spaces {
+        wrap_code_in_spaces(&state, checker, expr.range())
+    } else {
+        state.to_string()
+    };
+
+    Ok(Edit::range_replacement(code, expr.range()))
+}
+
+/// Adds spaces around the code generated from `state` if `wrap_in_spaces` is true.
+fn wrap_code_in_spaces(state: &CodegenState, checker: &Checker, expr_range: TextRange) -> String {
+    if checker.model.in_f_string() {
+        if let Some(f_string_range) = checker.indexer.f_string_range(expr_range.start()) {
+            let f_string = checker.locator.slice(f_string_range);
+
+            // find the braces that delimit the f-string
+            let f_string_start = f_string_range.start();
+            let f_string_sbrace =
+                f_string_start + TextSize::try_from(f_string.find('{').unwrap()).unwrap();
+            let f_string_ebrace =
+                f_string_start + TextSize::try_from(f_string.rfind('}').unwrap()).unwrap();
```

Comment:
(I'm being a bit vague because I don't know how exactly to solve this problem. We need to track additional state to detect the parent formatted expression.)

Reply:
oh, ok, guess I'll have to study the f-string literal syntax a bit more

Reply:
Ok, I see.

It seems I assumed (wrongly) that an expression wrapped in braces is always being formatted, but that's not the case.

Reply:
ok now I see your point, the code is wrong